odoo/addons/base/views/res_partner_views.xml

```diff
@@ -336,7 +336,7 @@
                         <page name='sales_purchases' string="Sales &amp; Purchase">
                             <group name="container_row_2">
                                 <group string="Sales" name="sale" priority="1">
-                                    <field name="user_id" domain="[('share', '=', False)]"/>
+                                    <field name="user_id" domain="[('share', '=', False), '|', ('company_id', '=?', company_id), ('company_ids', '=', company_id)]"/>
```

Comment:
``=?`` ?

Reply:
Return true if unset or None. And acts like a normal = if the value is set